pkg/api/oauthrelyingparty/provider.go

```diff
@@ -32,15 +32,26 @@ func (c ProviderConfig) ClientID() string {
 	return client_id
 }
 
-func (c ProviderConfig) SetDefaultsModifyDisabledFalse() {
-	_, ok := c["modify_disabled"].(bool)
+func (c ProviderConfig) SetDefaultsCreateDisabled(defaultValue bool) {
+	_, ok := c["create_disabled"].(bool)
 	if !ok {
-		c["modify_disabled"] = false
+		c["create_disabled"] = defaultValue
 	}
 }
 
-func (c ProviderConfig) ModifyDisabled() bool {
-	modify_disabled, _ := c["modify_disabled"].(bool)
+func (c ProviderConfig) SetDefaultsDeleteDisabled(defaultValue bool) {
+	_, ok := c["delete_disabled"].(bool)
+	if !ok {
+		c["delete_disabled"] = defaultValue
+	}
+}
+
+func (c ProviderConfig) CreateDisabled() bool {
+	modify_disabled, _ := c["create_disabled"].(bool)
+	return modify_disabled
+}
+func (c ProviderConfig) DeleteDisabled() bool {
+	modify_disabled, _ := c["delete_disabled"].(bool)
```

Comment:
ModifyDisabled is removed. But for the following case, it should be read.

The actual config
```yaml
modify_disabled: true
```
This project has configured modify_disabled=true for a particular provider.

Reply:
@louischan-oursky I think this part should be handled by the implementation of the provider.

Something like:

```go
func (ProviderABC) SetDefaults(cfg oauthrelyingparty.ProviderConfig) {
	deprecatedModifyDisabled, ok := cfg["modify_disabled"].(bool)
	if ok {
		cfg.SetDefaultsUpdateDisabled(deprecatedModifyDisabled)
	}
}
```

Reply:
According to our offline discussion, we will remove `Alias()`, `SetDefaultsModifyDisabledFalse()`, and `ModifyDisabled()`

Reply:
I pushed a new commit for this change, but I am still working on the authgear-server changes.